plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs';
+import path from 'path';
+import { promisify } from 'util';
+
+import {
+  GithubCredentialsProvider,
+  ScmIntegrationRegistry,
+} from '@backstage/integration';
+import { zipObject } from 'lodash';
+import { createTemplateAction } from '../../createTemplateAction';
+import { Octokit } from '@octokit/rest';
+import { OctokitResponse } from '@octokit/types';
+import { InputError } from '@backstage/errors';
+import { createPullRequest } from 'octokit-plugin-create-pull-request';
+import globby from 'globby';
+
+const readFile = promisify(fs.readFile);
+
+type PullRequestResponse = {
+  html_url: string;
+};
+
+export interface PullRequestCreator {
+  createPullRequest(
+    options: createPullRequest.Options,
+  ): Promise<OctokitResponse<PullRequestResponse> | null>;
```

Comment:
I think it would make sense to narrow this down to only the options that we actually pass in, and only include in the response what is actually being used. That way we get rid of the `@octokit/types` dependency, and it becomes easier to create a custom implementation of this.